ENIGMAsystem/SHELL/Universal_System/roomsystem.h

```diff
@@ -121,11 +121,11 @@ namespace enigma
   struct inst {
     int id,obj,x,y;
   };
-    struct tile {
-        int id,bckid,bgx,bgy,depth,height,width,roomX,roomY;
-        double alpha, xscale, yscale;
-        int color;
-    };
+  struct tile {
```

Comment:
you should probably define default values to the struct members or require a constructor incase someone forgets to set a field or a new feild is added later

Reply:
Sure, but we still haven't exactly decided on defaults for the protos yet either. This was Josh's doing, I'd have to refactor the whole file, and I don't see a "need" to just yet - though it can only help make it a little "safer". I may hold off on this one until another pr as well. Regardless, this is a valid point.

Reply:
Construction for this is emitted as generated code. Adding constructors could break the C-style initializer we're using right now, so use caution.